api/golang/engine/lib/kurtosis_context/kurtosis_context.go

```diff
@@ -60,12 +68,52 @@ func NewKurtosisContextFromLocalEngine() (*KurtosisContext, error) {
 	}
 
 	engineServiceClient := kurtosis_engine_rpc_api_bindings.NewEngineServiceClient(conn)
-	if err := validateEngineApiVersion(ctx, engineServiceClient); err != nil {
+	if err = validateEngineApiVersion(ctx, engineServiceClient); err != nil {
 		return nil, stacktrace.Propagate(err, "An error occurred validating the Kurtosis engine API version")
 	}
 
+	// TODO: Here we're checking whether the context is local or remote to connect to the daemon. This is because
+	//  the daemon will not immediately run on all Kurtosis users' laptop. Once that's the case, we can remove this if
+	//  and systematically connect to the locally running Kurtosis portal daemon.
+	currentContext, err := store.GetContextsConfigStore().GetCurrentContext()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "Error retrieving current context")
+	}
+	var portalClient portal_api.KurtosisPortalClientClient
+	instantiatePortalClientVisitor := contexts_store_api.KurtosisContextVisitor[struct{}]{
+		VisitLocalOnlyContextV0: func(localContext *contexts_store_generated_api.LocalOnlyContextV0) (*struct{}, error) {
+			portalClient = nil
+			return nil, nil
+		},
+		VisitRemoteContextV0: func(remoteContext *contexts_store_generated_api.RemoteContextV0) (*struct{}, error) {
+			// When the context is remote, we build a client to the locally running portal daemon
+			kurtosisPortalSocketStr := fmt.Sprintf("%v:%v", localHostIPAddressStr, DefaultGrpcPortalClientPortNum)
+			// TODO SECURITY: Use HTTPS to ensure we're connecting to the real Kurtosis API servers
+			portalConn, err := grpc.Dial(kurtosisPortalSocketStr, grpc.WithInsecure())
+			if err != nil {
+				return nil, stacktrace.Propagate(
+					err,
+					"An error occurred creating a connection to the Kurtosis Portal Client at '%v'",
+					kurtosisPortalSocketStr,
+				)
+			}
+			portalClient = portal_api.NewKurtosisPortalClientClient(portalConn)
+
+			if _, err = portalClient.Ping(context.Background(), portal_constructors.NewPortalPing()); err != nil {
+				return nil, stacktrace.Propagate(err, "Unable to connect to the locally running Kurtosis portal daemon."+
+					" To use remote context, a Kurtosis Portal daemon must be running on this machine")
```

Comment:
can we point people to how we they can run it?

Reply:
That's still a bit uncertain right now to be entirely honest with you 👼 👼 
So, until I upgrade the Portal to be runnable in a more productised way, I'll hold off adding more details here. But I can update once the portal run command looks better 👍